### DISCUSSION TITLE
现在我们的int float str ptr是小写开头的, 代码风格.

### DISCUSSION THREAD

**Discussion by zhangzhimin** [prajna-lang/prajna#26] [Programming Language Design]
是否应该使用Int Float String更为统一?

我们的Function名字开头也是小写? 是否改为大写更为合适?

**Comment by microctar** [comment]
内置类型(primitive type)小写开头合适，
自定义类型大写合适。

**Comment by microctar** [comment]
func依旧保持小写，也可进一步简写为fn， implement关键字同理。

**Reply by zhangzhimin** [reply]
Function名字我指的是 length(), cast这些

**Reply by microctar** [reply]
> Function名字我指的是 length(), cast这些

不必，snake_case就挺好。
可参考rust-lang。

如果寻求相近想法，可参考[carbon-lang: Naming conventions](https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0861.md)

**Reply by zhangzhimin** [reply]
我说的是isEmpty()这样的要不要开头大写为IsEmpty , 我们现在是小写开头

**Reply by microctar** [reply]
> 我说的是isEmpty()这样的要不要开头大写为IsEmpty , 我们现在是小写开头

可义考虑参照carbon-lang，如下：

Taking that into consideration, and that we are likely to keep lower_snake_case for keywords and type literals in particular, a couple options discussed would have mainly affect self, base, and string:

Anything language-provided without needing an import is always lower_snake_case.
i32, bool, true, self, base, string.is_empty()
As above, but prelude class members follow standard naming conventions.
i32, bool, true, self, base, string.IsEmpty()

我个人更偏向于IsEmpty，大写开头。
